### PR TITLE
Remove redundant conditional statement

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

1. remove redundant conditional statement
2. replace  errors.New(fmt.Sprintf with fmt.Errorf 

Thanks.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

File: pkg/kubectl/cmd/drain_test.go [hunk 1]
```diff
@@ -662,7 +661,7 @@ func TestDeletePods(t *testing.T) {
 				if oldPod, found := oldPodMap[name]; found {
 					return &oldPod, nil
 				}
-				return nil, errors.New(fmt.Sprintf("%q: not found", name))
+				return nil, fmt.Errorf("%q: not found", name)
```
Comment (resouer) [inline pkg/kubectl/cmd/drain_test.go]: This looks cleaner, thanks

File: pkg/kubectl/cmd/drain.go [hunk 2]
```diff
@@ -464,9 +464,9 @@ func (o *DrainOptions) deleteOrEvictPods(pods []api.Pod) error {
 
 	if len(policyGroupVersion) > 0 {
 		return o.evictPods(pods, policyGroupVersion, getPodFn)
-	} else {
-		return o.deletePods(pods, getPodFn)
 	}
+	return o.deletePods(pods, getPodFn)
```
Comment (resouer) [inline pkg/kubectl/cmd/drain.go]: I'm sorry, but could you please leave this and below as it is? I'm afraid the modification you made here makes the logic non-readable.

**Review by resouer — CHANGES_REQUESTED** [review]

**Comment by tanshanshan** [comment]
cc @resouer , PTAL

**Review by resouer — APPROVED** [review]

**Comment by tanshanshan** [comment]
> Adding do-not-merge because the release note process has not been followed.
One of the following labels is required "release-note", "release-note-action-required", "release-note-experimental" or "release-note-none".
Please see: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes.

cc @resouer , Could you give  one of these labels?

**Pull request merged**